### ISSUE TITLE
In Firefox: NVDA reads only half of the paragraph on hover/mouseover.

### ISSUE THREAD

**Issue by MoubinShiraz** [nvaccess/nvda#16902]
### Steps to reproduce:

1. Open Firefox browser 127.0 in windows 11 Home.
2. Open the codepen: https://codepen.io/Moubin-Shiraz/pen/LYKRpJB/
3. Hover on the text: **Please review your cart and provide any missing information. Proceed to fill the**

### Actual behavior:
NVDA doesn't announce/read the whole text: **Please review your cart and provide any missing information. Proceed to fill the**.
Instead it reads just: **Please review your cart and provide any miss**

### Expected behavior:
It should read the whole paragraph i.e: **Please review your cart and provide any missing information. Proceed to fill the** instead of reading it half.

### NVDA logs, crash dumps and other attachments:
There is no any error message.

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:

1. 2024.2 (2024.2.0.32555)
2. 2023.3.4

#### Windows version:
Windows 11 Home

#### Name and version of other software in use when reproducing the issue:
Firefox version: 127.0 & 128.0

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Same behaviour in 2 versions of NVDA listed above.

#### If NVDA add-ons are disabled, is your problem still occurring?
Yes

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by MoubinShiraz** [comment]
@hwf1324 maybe but not sure.

**Comment by XLTechie** [comment]
@MoubinShiraz Can you test in a Chromium browser please? Edge and/or Chrome?

@jcsteh in #15990, if this is the same, it is pointed out that this is only a problem with Firefox.

**Comment by MoubinShiraz** [comment]
@XLTechie FYI: the bug only occurs on Firefox browsers.

**Comment by jcsteh** [comment]
Best I can tell, this is not a Firefox bug. IA2_TEXT_BOUNDARY_PARAGRAPH is behaving exactly as it should. This looks like another bug in IA2TextTextInfo.expand, but I haven't worked out exactly what yet.

**Comment by jcsteh** [comment]
Ah, I see the problem. I have a patch which fixes it locally. I'll submit it when I get some time.

**Comment by MoubinShiraz** [comment]
@jcsteh Please submit it ASAP because our client needs it urgently,  that will be highly appreciated. Thanks.

**Comment by jcsteh** [comment]
If there were a paid contract involved, that would be a reasonable request. There isn't, so it isn't. I will submit it when I am able.

**Comment by jcsteh** [comment]
This is indeed a duplicate of #15990. Closing as such.
